src/libtego/source/context.cpp

```diff
@@ -0,0 +1,1030 @@
+#include "context.hpp"
+#include "error.hpp"
+#include "globals.hpp"
+#include "tor.hpp"
+#include "user.hpp"
+#include "ed25519.hpp"
+
+using tego::g_globals;
+
+#include "tor/TorControl.h"
+#include "tor/TorManager.h"
+#include "tor/TorProcess.h"
+#include "core/UserIdentity.h"
+#include "core/ContactUser.h"
+#include "core/ConversationModel.h"
+
+//
+// Tego Context
+//
+
+tego_context::tego_context()
+: callback_registry_(this)
+, callback_queue_(this)
+{
+    this->torManager = Tor::TorManager::instance();
+    this->torControl = torManager->control();
+}
+
+void tego_context::start_tor(const tego_tor_launch_config_t* config)
+{
+    TEGO_THROW_IF_NULL(config);
+
```

Comment:
should there be a `TEGO_THROW_IF_NULL(this->torManager);` here like in `get_tor_daemon_configured()`?